src/helper.h

```diff
@@ -245,27 +233,29 @@ class helper_server : public HelperServerBase
     CBDATA_CHILD(helper_server);
 
 public:
-    uint64_t nextRequestId;
+    helper_server(helper *hlp, int pid, void *hIpc, int rfd, int wfd);
```

Comment:
FYI: Ideally, we should avoid adding function calls with a large number of parameters. We should especially avoid function calls with easy-to-mix-up parameters (like rfd and wfd here). It is better to set each such members explicitly, by name. You may recall multiple bugs caused by interchangeable eventAdd() parameters.

It is a very difficult problem to solve properly, and I do not insist on fixing it in this context, but this PR does create it (while fixing many other problems).

Reply:
The initialization bugs I referred to were members of parent class up the hierarchy failing to be initialized correctly (at all) - and others being double-assigned with different values from the caller vs constructor sequence. Having the initialization outside the constructor is exactly what lead to those existing.

Some of the stats values may have been noticed but the stats gathering logic blamed when it was the broken ctor sequence initializing a wrong value.

Reply:
> The initialization bugs I referred to were members of parent class up the hierarchy failing to be initialized correctly (at all) - and others being double-assigned with different values from the caller vs constructor sequence. Having the initialization outside the constructor is exactly what lead to those existing.

I disagree with you analysis: Lack of initialization is certainly a bug, but it is not caused by post-construction object configuration. And "double-assignment" is not a bug per se. However, this disagreement is currently unimportant AFAICT.

Configuring objects with a large number of parameters is a difficult problem to solve, and no C++11 solution (that I know of) is perfect. Your solution is acceptable IMO. I am resolving this thread.